File: sample/src/main/java/com/example/kyrie/SampleOnClickListener.java
package com.example.kyrie;

import android.support.annotation.NonNull;
import android.view.View;

import com.github.alexjlockwood.kyrie.KyrieDrawable;

final class SampleOnClickListener implements View.OnClickListener {
  @NonNull private final KyrieDrawable drawable;

  public SampleOnClickListener(@NonNull KyrieDrawable drawable) {
    this.drawable = drawable;
  }

  @Override
  public void onClick(View v) {
    if (drawable.isPaused()) {
      drawable.resume();
    } else {
      if (drawable.isStarted()) {
        drawable.pause();
      } else {
        drawable.start();
      }
    }
  }
}

File: sample/src/main/java/com/example/kyrie/SampleOnClickListener.kt
package com.example.kyrie

import android.view.View

import com.github.alexjlockwood.kyrie.KyrieDrawable

internal class SampleOnClickListener(private val drawable: KyrieDrawable) : View.OnClickListener {

    override fun onClick(v: View) {
        if (drawable.isPaused) {
            drawable.resume()
        } else {
            if (drawable.isStarted) {
                drawable.pause()
            } else {
                drawable.start()
            }
        }
    }
}
